### PR TITLE
KTOR-1492 Improve exception messages and fix grammar mistakes

### PR DESCRIPTION
**Pull request by hhariri**

### CODE REVIEW

File: ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt [hunk 1]
```diff
@@ -103,7 +103,7 @@ public class JavaHttpEngine(override val config: JavaHttpConfig) : HttpClientEng
                 proxy(ProxySelector.of(address))
             }
             Proxy.Type.DIRECT -> proxy(HttpClient.Builder.NO_PROXY)
-            else -> throw IllegalStateException("Proxy of type $type is unsupported by Java HTTP engine.")
+            else -> throw IllegalStateException("Java HTTP engine does not currently support $type proxies.")
```
Comment (Stexxe) [inline ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt:106]: Why do we need the word `currently`?

File: ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt [hunk 2]
```diff
@@ -54,7 +54,7 @@ public class CIOApplicationEngine(environment: ApplicationEngineEnvironment, con
         try {
             environment.connectors.forEach { connectorSpec ->
                 if (connectorSpec.type == ConnectorType.HTTPS) {
-                    throw UnsupportedOperationException("HTTPS is not supported by CIO engine")
+                    throw UnsupportedOperationException("CIO Engine does not currently support HTTPS. Please consider using a different engine if you require HTTPS")
```
Comment (Stexxe) [inline ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt:57]: > Please consider using a different engine if you require HTTPS

Isn't that implied from the first sentence?

File: ktor-server/ktor-server-tomcat/jvm/src/io/ktor/server/tomcat/TomcatApplicationEngine.kt [hunk 3]
```diff
@@ -74,7 +74,7 @@ public class TomcatApplicationEngine(environment: ApplicationEngineEnvironment,
                             scheme = "https"
 
                             if (ktorConnector.keyStorePath == null) {
-                                throw IllegalArgumentException("Tomcat requires keyStorePath")
+                                throw IllegalArgumentException("Tomcat requires keyStorePath. Make sure you're setting the property in the EngineSSLConnectorConfig class used")
```
Comment (Stexxe) [inline ktor-server/ktor-server-tomcat/jvm/src/io/ktor/server/tomcat/TomcatApplicationEngine.kt:77]: What does the word `used` belong to?

File: ktor-utils/common/src/io/ktor/util/StringValues.kt [hunk 4]
```diff
@@ -251,7 +251,7 @@ public open class StringValuesBuilder(public val caseInsensitiveName: Boolean =
 
     private fun ensureListForKey(name: String, size: Int): MutableList<String> {
         if (built) {
-            throw IllegalStateException("Cannot modify a builder when final structure has already been built")
+            throw IllegalStateException("Cannot modify a builder after build() function already invoked. Make sure you call build() last.")
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/StringValues.kt:254]: > Make sure you call build() last

What does this mean?

File: ktor-utils/jvm/src/io/ktor/util/NioPath.kt [hunk 5]
```diff
@@ -22,7 +22,7 @@ public val Path.extension: String get() = fileName.toString().substringAfterLast
 public fun Path.combineSafe(relativePath: Path): File {
     val normalized = relativePath.normalizeAndRelativize()
     if (normalized.startsWith("..")) {
-        throw InvalidPathException(relativePath.toString(), "Bad relative path $relativePath")
+        throw InvalidPathException(relativePath.toString(), "Relative path $relativePath starts with .., which is invalid")
```
Comment (Stexxe) [inline ktor-utils/jvm/src/io/ktor/util/NioPath.kt:25]: Maybe, we should rephrase it to `Relative path beginning with .. is invalid` or something like this

File: ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt (hunk 1 above)
Comment (hhariri) [reply ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt:106]: Because at some point we'll support it, and it's to indicate that this isn't permanent.

File: ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt (hunk 2 above)
Comment (hhariri) [reply ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt:57]: It's providing a helpful step as to what to do.

File: ktor-utils/common/src/io/ktor/util/StringValues.kt (hunk 4 above)
Comment (hhariri) [reply ktor-utils/common/src/io/ktor/util/StringValues.kt:254]: The build() function of the buider.

**Review by e5l — APPROVED** [review]
lgtm

File: ktor-utils/jvm/src/io/ktor/util/NioPath.kt (hunk 5 above)
Comment (hhariri) [reply ktor-utils/jvm/src/io/ktor/util/NioPath.kt]: Will do.
